yarn-project/simulator/src/avm/journal/trace.ts

```diff
@@ -169,4 +170,14 @@ export class WorldStateAccessTrace {
     // it is assumed that the incoming trace was initialized with this as parent, so accept counter
     this.accessCounter = incomingTrace.accessCounter;
   }
+
+  // TODO(dbanks12): should only return hints for one call.... shouldn't include nested calls (merged in traces)
+  public toHints(): AvmExecutionHints {
+    return new AvmExecutionHints(
+      this.publicStorageReads.map(read => new AvmHint(read.slot, read.value)),
+      this.noteHashChecks.map(check => new AvmHint(check.noteHash, new Fr(check.exists))),
+      this.nullifierChecks.map(check => new AvmHint(check.nullifier, new Fr(check.exists))),
+      this.l1ToL2MessageChecks.map(check => new AvmHint(check.msgHash, new Fr(check.exists))),
```

Comment:
I thought about these ones a bit more. Could you explicitly do `check.exists ? new Fr(1) : Fr.ZERO`? I wouldn't trust C++ for autoconversions of booleans (anything non-zero is true).